src/AutoMapper/ConfigurationStore.cs

```diff
@@ -417,10 +417,15 @@ public bool HasOpenGenericTypeMapDefined(ResolutionContext context)
 
         private IEnumerable<TypePair> GetRelatedTypePairs(TypePair root)
         {
-            return
+            var includeOverrideTypePairs =
+                GetAllTypeMaps()
+                    .Where(tm => tm.SourceType.IsAssignableFrom(root.SourceType) && (tm.DestinationTypeOverride ?? tm.DestinationType) != root.DestinationType && tm.DestinationType.IsAssignableFrom(root.DestinationType))
+                    .Select(tm => new TypePair(tm.SourceType,tm.DestinationTypeOverride ?? tm.GetDerivedTypeFor(root.SourceType))).ToList();
```

Comment:
Actually there's some problems with this in inheritance and #842.  It's quite difficult to determine what is inherited and what is not.